### PR TITLE
feat: [STO-5353]: Add RBAC for Ticket Service

### PR DESCRIPTION
**Pull request by chris-harness**
Adds new RBAC for the new Ticket Service.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42399)

### CODE REVIEW

**Comment by chris-harness** [comment]
trigger messagecheck

**Review by r0610205 — APPROVED** [review]

**Review by atmollo-harness — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42399)* status: :shipit: complete! ✓

**Comment by chris-harness** [comment]
trigger messagecheck

**Comment by chris-harness** [comment]
trigger messagecheck

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42399)* status: :shipit: complete! ✓

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42399)* status: 3 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml` line 4 at r3](https://reviewable.io/reviews/harness/harness-core/42399#-NLdbSNSDhu0F7FmDX4-:-NLdbSNSDhu0F7FmDX40:b-fqf43i) ([raw file](https://github.com/harness/harness-core/blob/1ca73c5e70d9479e861e7951a96e6f557a869d47/access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml#L4)):*
> ```YAML
> name: rolesConfiguration
> version: 43
> roles:
> ```

Can u rebase the latest versions is 41. So new version should be 42.

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Any edit in this file should accompany an increment in version.
 name: rolesConfiguration
-version: 41
+version: 43
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:3]: 42?

**Review by reetika04 — CHANGES_REQUESTED** [review]

**Review by Jimit09Gandhi — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42399#-:-NLfTkaM8y51B4f0yCyC:bnfp4nl)* status: :shipit: complete! ✓

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml (hunk 1 above)
Comment (chris-harness) [reply access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml]: Fixed

**Pull request merged**